### PR TITLE
Assignments layout and fixes

### PR DESCRIPTION
**Pull request by MaximilianKrambach**
See #4558

### CODE REVIEW

File: client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html [hunk 1]
```diff
@@ -74,27 +68,32 @@ <h1>{{ assignment.title }}</h1>
 </ng-template>
 
 <ng-template #metaInfoTemplate>
+    <h2> {{ assignment.title }}</h2>
```
Comment (FinnStutzenstein) [inline client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html]: `.getTitle()`

**Comment by MaximilianKrambach** [comment]
A bit of functionality fxes sneaked in here, too:
- permission fix for non-admin users
- fix one-vote polls (pollmethod: 'votes')
- The candidate list will now be filtered

File: client/src/app/core/repositories/assignments/assignment-repository.service.ts [hunk 2]
```diff
@@ -210,7 +210,8 @@ export class AssignmentRepositoryService extends BaseAgendaContentObjectReposito
      * @param assignment
      */
     public async sortCandidates(sortedCandidates: any[], assignment: ViewAssignment): Promise<void> {
-        throw Error('TODO');
+        throw new Error('TODO');
+        // TODO responds with 405 method not allowed:
         // const restPath = `/rest/assignments/assignment/${assignment.id}/sort_related_users`;
         // const data = { related_users: sortedCandidates };
         // await this.httpService.post(restPath, data);
```
Comment (MaximilianKrambach) [inline client/src/app/core/repositories/assignments/assignment-repository.service.ts]: @FinnStutzenstein The server does not seem to like this request. Could you please take a look?

**Comment by MaximilianKrambach** [comment]
@GabrielInTheWorld @FinnStutzenstein  The code reused from 'List of speakers' seems to introduce quite a performance drop right now (with my filteredCandidates instead of a full availableCandidates). Maybe it's something quite obvious (I introduced this part in https://github.com/OpenSlides/OpenSlides/pull/4577/commits/2101478266273a173cba795074814f5f154f9177)

**Comment by tsiegleauq** [comment]
Tested. Does not seem to break anything and the most stuff seems to work fine.

**Pull request merged**